lib/model.js

```diff
@@ -3948,7 +3948,7 @@ function getModelsMapForPopulate(model, docs, options) {
     let ret;
 
     if (localFieldGetters.length) {
-      ret = localFieldPath.applyGetters(doc[localField], doc);
+      ret = localFieldPath.applyGetters(doc[localField], model.hydrate(doc));
```

Comment:
Only suggestion would be to check if `doc` is already hydrated before hydrating. Generally we use `doc.$__ != null` to check whether the doc is already hydrated.